src/Gold.Redis/Gold.Redis.LowLevelClient/Communication/RedisSocketAuthenticator.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Gold.Redis.Common.Interfaces.Communication;
+using System;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace Gold.Redis.LowLevelClient.Communication
+{
+    public class RedisSocketAuthenticator : IRedisAuthenticator
+    {
+        private readonly ISocketCommandExecutor _socketCommandExecutor;
+        public RedisSocketAuthenticator(ISocketCommandExecutor socketCommandExecutor)
+        {
+            _socketCommandExecutor = socketCommandExecutor;
+        }
+        public async Task<bool> TryAuthenticate(Socket connectionSocket, string password)
+        {
+            try
+            {
+                var authCommand = "AUTH " + password;
+                var response = await _socketCommandExecutor.ExecuteCommand(connectionSocket, authCommand);
+                return response == "OK";

```

Comment:
OK is a global redis response for a lot of things. It dosent need to be hard-coded here

Reply:
Fixed -> moved to Constant class